### PR TITLE
SWARM-730 - Public JavaDocs

### PR DESCRIPTION
**Pull request by bobmcwhirter**
- [X] Have you followed the guidelines in our [Contributing](http://wildfly-swarm.io/community/contributing/) document?
- [X] Have you created a [JIRA](https://issues.jboss.org/browse/SWARM) and used it in the commit message?
- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/wildfly-swarm/wildfly-swarm/pulls) for the same issue?
- [X] Have you built the project locally prior to submission with `mvn clean install`?

*PARTIAL FIX FOR SWARM-730 -- DO NOT RESOLVE IN JIRA*
-----

Begin documenting public-facing interfaces some.

### CODE REVIEW

File: arquillian/adapter/src/main/java/org/wildfly/swarm/arquillian/adapter/AnnotationBasedMain.java [hunk 1]
```diff
@@ -50,7 +50,7 @@ public static void main(String... args) throws Exception {
             }
 
             boolean startEagerly = anno.startEagerly();
```
Comment (kenfinnigan) [inline arquillian/adapter/src/main/java/org/wildfly/swarm/arquillian/adapter/AnnotationBasedMain.java:52]: Remove this  too?

File: core/container/src/main/java/org/wildfly/swarm/container/Container.java [hunk 2]
```diff
@@ -70,12 +70,6 @@ public Container fraction(Fraction fraction) {
```
Comment (kenfinnigan) [inline core/container/src/main/java/org/wildfly/swarm/container/Container.java:70]: Are we at the point yet where we can actually remove this class entirely?

**Review by kenfinnigan — APPROVED** [review]

**Comment by Ladicek** [comment]
Love this!

File: core/spi/src/main/java/org/wildfly/swarm/spi/api/SocketBinding.java [hunk 3]
```diff
@@ -15,7 +15,15 @@
  */
 package org.wildfly.swarm.spi.api;
 
-/**
+/** An inbound socket-binding.
+ *
+ * <p>Defines a name inbound normal or multicast socket-binding.</p>
```
Comment (Ladicek) [inline core/spi/src/main/java/org/wildfly/swarm/spi/api/SocketBinding.java]: named

File: core/container/src/main/java/org/wildfly/swarm/Swarm.java [hunk 4]
```diff
@@ -215,19 +242,53 @@ public Swarm(boolean debugBootstrap, String... args) throws Exception {
...
+     * Pass the effective command-line arguments to this instance.
+     *
+     * @param args The arguments.
+     */
     public void setArgs(String... args) {
         this.args = args;
     }
 
+    /**
+     * Specify an XML configuration file (in usualy WildFly {@code standalone.xml} format.
```
Comment (Ladicek) [inline core/container/src/main/java/org/wildfly/swarm/Swarm.java]: usual (I think), closing paren missing

File: core/container/src/main/java/org/wildfly/swarm/Swarm.java [hunk 5]
```diff
@@ -265,11 +336,31 @@ public Swarm component(Class<?> cls) {
         return this;
     }
 
+    /**
+     * Add an outbound socket-binding to the container.
+     *
+     * <p>In the even the specified {@code socketBindingGroup} does no exist, the socket-binding
```
Comment (Ladicek) [inline core/container/src/main/java/org/wildfly/swarm/Swarm.java]: even -> event, does no exist -> does not exist

This also appears in the `socketBinding` method.

**Comment by Ladicek** [comment]
These are all the typos I found. Otherwise +999 to this.

**Comment by bobmcwhirter** [comment]
Will fix typos before merge.

File: arquillian/adapter/src/main/java/org/wildfly/swarm/arquillian/adapter/AnnotationBasedMain.java (hunk 1 above)
Comment (bobmcwhirter) [reply arquillian/adapter/src/main/java/org/wildfly/swarm/arquillian/adapter/AnnotationBasedMain.java:52]: Probably, but in a separate PR.

File: core/container/src/main/java/org/wildfly/swarm/container/Container.java (hunk 2 above)
Comment (bobmcwhirter) [reply core/container/src/main/java/org/wildfly/swarm/container/Container.java:70]: Also probably, but also in a separate PR.

**Pull request merged**